src/foam/lang/types.js

```diff
@@ -195,32 +206,39 @@ foam.CLASS({
     },
     {
       name: 'adapt',
-      value: function (_, d) {
+      value: function (_, d, prop) {
         if ( d === undefined || d === null ) return d;
         var originalDate = d;
         if ( typeof d === 'number' )
           d = new Date(d);
 
         if ( typeof d === 'string' ) {
-          var match = d.match(/^(\d{4})[-\/]?(\d{2})[-\/]?(\d{2})$/);
-
-          if ( match ) {
-            // Parse yyyy-mm-dd, yyyy/mm/dd, or yyyymmdd formats
-            // T00:00:00 is required to force local timezone interpretation
-            // Without it, new Date("2024-03-15") is parsed as UTC midnight
-            // which displays as previous day in negative UTC offset timezones
-            d = new Date(match[1] + '-' + match[2] + '-' + match[3] + 'T00:00:00');
-          } else {
-            // Parse mm/dd/yyyy
-            match = d.match(/^(\d{2})[-\/]?(\d{2})[-\/]?(\d{4})$/);
+          // Try each format in formatsOrder until one matches
```

Comment:
Would be better to just call DateUtil.parseDateString() and so we can standardize to one location rather than duplicating here, in CSV, JSON, XML, etc.